### PR TITLE
Add timepicker

### PR DESCRIPTION
**Pull request by Edward-Phillips**
# Changes

 - adds timePicker from DateTimePicker library - https://docs.expo.dev/versions/latest/sdk/date-time-picker/
 - adds mobx state values for storing/setting alarm time and toggling whether or not to display the time picker
 - extracts `FormattedTimeDisplay` to own component to avoid re-rendering issue when formatted time changes each second

### CODE REVIEW

File: app/(tabs)/timeScreen.tsx [hunk 1]
```diff
@@ -4,22 +4,38 @@ import useViewModel from '@/viewModels/useViewModel'
...
 
 
 export default observer(function timeScreen ()  {
   const viewModel = useViewModel(timeScreenVM)
 
-  const { formattedTime, clickCount, incrementClickCount, dispose } = viewModel
-
-  useEffect(() => {
-    return () => dispose()
-  }, [viewModel])
```
Comment (AlecDWalker) [inline app/(tabs)/timeScreen.tsx]: useEffect may not have needed removal?

File: components/FormattedTimeDisplay.tsx
Comment (AlecDWalker) [inline components/FormattedTimeDisplay.tsx:1]: Nice and straightforward

**Review by AlecDWalker — COMMENTED** [review]
Looks good to me

File: app/(tabs)/timeScreen.tsx (hunk 1 above)
Comment (Edward-Phillips) [reply app/(tabs)/timeScreen.tsx]: good catch! I'll fix that now

**Review by AlecDWalker — APPROVED** [review]

**Pull request merged**